data_preparation/raster.py

```diff
@@ -0,0 +1,153 @@
+# File: raster.py
+# Author: Joint Research Center
+
+import numpy as np
+import rasterio
+import rasterio.features
+
+
+def read_masked(ds, mask, indexes=None, window=None,
+                use_pixels='CENTER', out_mask_path=None,
+                *args, **kwargs):
+    """
+    Reads the data fom the raster file and returns it as a numpy masked array.
+    The geometry is used to define the mask.
+    It returns only a subset of the data.
+        It reads only the raster window containing the geometry. This optimizes the performance by reducing disk reads.
+    The "use_pixels" parameter can be used to define how to handle the border pixels.
+        Some require shapely to be available on the system
+    :param ds: Raster file path or an instance of a rasterio.DatasetReader (already opened raster file).
+    :param mask: geojson geometry used to mask the array, mask is a read mask (only data covered by the geometry is used)

```

Comment:
The description should be updated to match the code change from the original. "mask" changed form geometry ->  iterable over geometries